spec/lib/pdf_filler/nj1040_pdf_spec.rb

```diff
@@ -1225,6 +1225,78 @@
       end
     end
 
+    describe "line 28a - Pension/Retirement Exclusion" do
+      before do
+        allow(Flipper).to receive(:enabled?).with(:show_retirement_ui).and_return(true)
+      end
+      
+      context "when taxpayer has pension/retirement income exclusion with 123_456" do
+        it "fills in the PDF line 28a boxes with the rounded value" do
+          allow_any_instance_of(Efile::Nj::Nj1040Calculator).to receive(:calculate_line_28a).and_return 123_456
+
+          # thousands
+          expect(pdf_fields["28a"]).to eq "1"
+          expect(pdf_fields["189"]).to eq "2"
+          expect(pdf_fields["190"]).to eq "3"
+          # hundreds
+          expect(pdf_fields["undefined_81"]).to eq "4"
+          expect(pdf_fields["191"]).to eq "5"
+          expect(pdf_fields["192"]).to eq "6"
+          # decimals
+          expect(pdf_fields["undefined_82"]).to eq "0"
+          expect(pdf_fields["193"]).to eq "0"
+        end
+      end
+    end
+
+    describe "line 28b - Other Retirement Income Exclusion" do
+      before do
+        allow(Flipper).to receive(:enabled?).with(:show_retirement_ui).and_return(true)
+      end
+      
+      context "when taxpayer has exclusion with 123_456" do
+        it "fills in the PDF line 28b boxes with the rounded value" do
+          allow_any_instance_of(Efile::Nj::Nj1040Calculator).to receive(:calculate_line_28b).and_return 123_456
+
+          # thousands
+          expect(pdf_fields["28b"]).to eq "1"
+          expect(pdf_fields["194"]).to eq "2"
+          expect(pdf_fields["195"]).to eq "3"
+          # hundreds
+          expect(pdf_fields["undefined_83"]).to eq "4"
+          expect(pdf_fields["196"]).to eq "5"
+          expect(pdf_fields["197"]).to eq "6"
+          # decimals
+          expect(pdf_fields["undefined_84"]).to eq "0"
+          expect(pdf_fields["198"]).to eq "0"
+        end
+      end
+    end
+
+    describe "line 28c - Total Retirement Income Exclusion" do
+      before do
+        allow(Flipper).to receive(:enabled?).with(:show_retirement_ui).and_return(true)
+      end
+      
+      context "when taxpayer has total retirement income exclusion with 123_456" do
+        it "fills in the PDF line 28c boxes with the rounded value" do
+          allow_any_instance_of(Efile::Nj::Nj1040Calculator).to receive(:calculate_line_28c).and_return 123_456
+
+          # thousands
+          expect(pdf_fields["28c"]).to eq "1"
+          expect(pdf_fields["199"]).to eq "2"
+          expect(pdf_fields["200"]).to eq "3"
+          # hundreds
+          expect(pdf_fields["undefined_85"]).to eq "4"
+          expect(pdf_fields["201"]).to eq "5"
+          expect(pdf_fields["202"]).to eq "6"
+          # decimals
+          expect(pdf_fields["undefined_86"]).to eq "0"
+          expect(pdf_fields["203"]).to eq "0"
+        end
```

Comment:
[pebble] could we test the empty case as well for all of these? When the number is zero, all the pdf fields should be blank